client/ayon_deadline/plugins/publish/global/collect_jobinfo.py

```diff
@@ -0,0 +1,351 @@
+            ),
+            EnumDef(
+                "primary_pool",
+                label="Primary pool",
+                default="none",
+                items=cls.pool_enum_values,
+            ),
+            EnumDef(
+                "secondary_pool",
+                label="Secondary pool",
+                default="none",
+                items=cls.pool_enum_values,
+            ),
+            EnumDef(
+                "machine_list",
+                label="Machine list",
+                multiselection=True,
+                default=default_values.get("machine_list"),
+                items=cls.machines_enum_values,
+            ),
+            BoolDef(
+                "machine_list_deny",
+                label="Machine List is a Deny",
+                default=default_values.get("machine_list_deny")
+            ),
+            TextDef(
+                "job_delay",
+                label="Delay job",
+                default=default_values.get("job_delay"),
+                tooltip="Delay job by specified timecode. Format: dd:hh:mm:ss",
+                placeholder="00:00:00:00"
+            )
+        ]
+
+        return [
+            attr_def
+            for attr_def in attr_defs
+            if attr_def.key in overrides
+        ]
+
+    @classmethod
+    def register_create_context_callbacks(cls, create_context):
+        create_context.add_value_changed_callback(cls.on_values_changed)
+
+    @classmethod
+    def on_values_changed(cls, event):
+        for instance_change in event["changes"]:
+            instance = instance_change["instance"]
+            #recalculate only if context changes
+            if "productName" not in instance_change:
+                continue
+
+            if not cls.instance_matches_plugin_families(instance):
+                continue
+
+            new_attrs = cls.get_attr_defs_for_instance(
+                event["create_context"], instance
+            )
+            instance.set_publish_plugin_attr_defs(cls.__name__, new_attrs)
+
+    def _get_jobinfo_defaults(self, instance):
+        """Queries project setting for profile with default values
+
+        Args:
+            instance (pyblish.api.Instance): Source instance.
+
+        Returns:
+            (dict)
+        """
+        context_data = instance.context.data
+        host_name = context_data["hostName"]
+        task_entity = context_data["taskEntity"]
+
+        task_name = task_type = None
+        if task_entity:
+            task_name = task_entity["name"]
+            task_type = task_entity["taskType"]
+
+        profile = filter_profiles(
+            self.profiles,
+            {
+                "host_names": host_name,
+                "task_types": task_type,
+                "task_names": task_name,
+                # "product_type": product_type
+            }
+        )
+        return profile or {}
+
+    @classmethod
+    def _host_specific_attr_defs(cls, create_context, instance, defs):
```

Comment:
This should either just return host specific attr defs, or not return defs.

Current implementation hits: "You can change definitions that are not host specific", which we probably don't want to? They are not used for anything...

```python
    @classmethod
    def _host_specific_attr_defs(cls, create_context, instance):
        host_name = create_context.host_name
        if host_name == "maya":
            return [
                NumberDef(
                    "tile_priority",
                    label="Tile Assembler Priority",
                    decimals=0,
                ),
                BoolDef(
                    "strict_error_checking",
                    label="Strict Error Checking",
                ),
            ]

        return []
```

Reply:
Updated.